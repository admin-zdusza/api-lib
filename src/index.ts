export { getGotClient } from "./service/utils/get-got-client";
export { getPostgresClient } from "./service/utils/get-postgres-client";
export { createLogger, logStorableEvent, logInfo, logError } from "./service/utils/logger";
export { staticHandler } from "./service/middlewares/static-handler";
export { defaultHandler } from "./service/middlewares/default-handler";
