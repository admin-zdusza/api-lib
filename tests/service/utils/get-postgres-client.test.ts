import { getPostgresClient } from "../../../src";
import { GenericContainer } from "testcontainers";

let container;

beforeAll(async () => {
  jest.setTimeout(300000);
  container = await new GenericContainer("postgres:10.16")
    .withEnv("POSTGRES_USER", "test")
    .withEnv("POSTGRES_PASSWORD", "test")
    .withExposedPorts(5432)
    .start();
});

afterAll(async () => {
  await container.stop();
  jest.setTimeout(5000);

});

test("should get postgres client", async () => {
  const dbUrl = `postgres://test:test@localhost:${container.getMappedPort(
    5432
  )}/test`;
  const db = await getPostgresClient(dbUrl, false, false);
  let response = await db.raw("select 1.2 as test_value_1, 1.2::float as test_value_2");
  expect(response.rows).toMatchSnapshot();
  await db.schema.createTable('test', function (table) {
    table.increments();
    table.string('name');
  });
  const insertResult = await db('test').insert([{name: 'test'}, {name: 'test2'}]).returning('*');
  expect(insertResult).toMatchSnapshot();
  response = await db.raw("select * from test");
  expect(response.rows).toMatchSnapshot();
});


test("should get postgres client with ssl", async () => {
  await expect(async () => {
    const dbUrl = `postgres://test:test@localhost:${container.getMappedPort(
      5432
    )}/test`;
    const db = await getPostgresClient(dbUrl, true, false);
    await db.raw('SELECT 1');
  })
    .rejects
    .toThrow('The server does not support SSL connections');
});
