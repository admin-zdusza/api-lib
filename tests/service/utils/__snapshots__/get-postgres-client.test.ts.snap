// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get postgres client 1`] = `
Array [
  Object {
    "testValue_1": 1.2,
    "testValue_2": 1.2,
  },
]
`;

exports[`should get postgres client 2`] = `
Array [
  Object {
    "id": 1,
    "name": "test",
  },
  Object {
    "id": 2,
    "name": "test2",
  },
]
`;

exports[`should get postgres client 3`] = `
Array [
  Object {
    "id": 1,
    "name": "test",
  },
  Object {
    "id": 2,
    "name": "test2",
  },
]
`;
